### PR TITLE
fix: range filter min/max displays -infinity when actual minimum is 0 (#889)

### LINKED ISSUE #889: Range filter min/max displays -Infinity when actual minimum is 0
## Problem

Range filters display `Min allowed: -Infinity` when the actual minimum value in the column is `0`. This is because the fallback logic uses `||` (logical OR) instead of `??` (nullish coalescing), and `0 || -Infinity` evaluates to `-Infinity` since `0` is falsy in JavaScript.

## Root Cause

Two locations use `||` to provide fallback values for min/max:

**`src/components/Table/common/utils.ts:55-56`**
```ts
max: minMax?.[1] || Infinity,
min: minMax?.[0] || -Infinity,
```

**`src/components/Filter/components/adapters/tanstack/ColumnFiltersAdapter/utils.ts:187-188`**
```ts
max: minMax?.[1] || Infinity,
min: minMax?.[0] || -Infinity,
```

`getFacetedMinMaxValues()` correctly returns `[0, 61.31]`, but `0 || -Infinity` gives `-Infinity`.

## Fix

Replace `||` with `??` in both locations:

```ts
max: minMax?.[1] ?? Infinity,
min: minMax?.[0] ?? -Infinity,
```

`??` only falls back on `null`/`undefined`, not on `0`.

## Reproduction

Any column with `filterFn: "inNumberRange"` where the actual minimum value is `0` will display `-Infinity` in the range filter UI. Observed on the HPRC Data Explorer Sequencing Data tab with the Coverage column.

### PR DESCRIPTION
**Pull request by frano-m**
## Summary
- Fixes range filter displaying `Min allowed: -Infinity` when the actual minimum value is `0`, caused by `||` treating `0` as falsy
- Extracts `getRangeMin`/`getRangeMax` helper functions using `??` (nullish coalescing) instead of `||`
- Fixes `buildRangeCategoryView` to disable the range filter when all values are non-numeric (min/max are both infinite)

### Files changed
- `src/common/categories/views/range/utils.ts` — Added `getRangeMin`/`getRangeMax` helpers; `buildRangeCategoryView` now sets `isDisabled` when both min and max are infinite
- `src/components/Table/common/utils.ts` — Replaced inline `||` fallbacks with `getRangeMin`/`getRangeMax`
- `src/components/Filter/components/adapters/tanstack/ColumnFiltersAdapter/utils.ts` — Same replacement
- `tests/getRangeMinMax.test.ts` — 8 tests covering `0` edge cases and `undefined` fallbacks

Closes #889

## Test plan
- [x] All 363 existing tests pass
- [x] 8 new unit tests for `getRangeMin`/`getRangeMax`
- [x] Manually verified on HPRC Data Explorer: Coverage filter shows `Min allowed: 0` instead of `-Infinity`
- [x] Manually verified range filter disables when filtered results have only non-numeric values

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/common/categories/views/range/utils.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function buildRangeCategoryView(
   );
   return {
     annotation: categoryConfig?.annotation,
-    isDisabled: false,
+    isDisabled: !isFinite(category.min) && !isFinite(category.max),
```
[the remark this answers is not in the document]
Comment (frano-m) [reply src/common/categories/views/range/utils.ts]: Good call — switched to `Number.isFinite` to avoid coercion.

File: src/common/categories/views/range/utils.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export function buildRangeCategoryView(
...
     key: category.key,
```
[the remark this answers is not in the document]
Comment (frano-m) [reply src/common/categories/views/range/utils.ts:28]: Added — three tests covering the disabled/enabled cases for `buildRangeCategoryView`: infinite min/max disables, finite values enable, and min of 0 with finite max enables.

**Review by NoopDog — APPROVED** [review]
